client/core/core.go

```diff
@@ -2016,66 +2017,26 @@ func (c *Core) ReconfigureWallet(appPW, newWalletPW []byte, assetID uint32, cfg
 	// If there are active trades, make sure they can be settled by the
 	// keys held within the new wallet.
 	sameWallet := func() error {
-		ownsAddr := func(addr string) error {
-			owns, err := wallet.OwnsAddress(addr)
-			if err != nil {
-				return err
-			}
-			if !owns {
-				return fmt.Errorf("new wallet does not own address found in active trades: %v", addr)
-			}
-			return nil
-		}
+		hasActiveTrade := false
 		for _, dc := range c.dexConnections() {
-			maybeDifferentWallet := false
 			for _, trade := range dc.trackedTrades() {
 				if !trade.isActive() {
 					continue
 				}
 				waID := wallet.AssetID
-				// If the to asset, check if we own an
-				// contract.Address.
-				if trade.wallets.toAsset.ID == waID {
-					if err := ownsAddr(trade.Trade().SwapAddress()); err != nil {
-						return err
-					}
-					// Assume all trade addresses are
-					// owned by the new wallet if one is.
-					return nil
-				}
-				// If the from asset, check if we own a
-				// refund address for a match if any exist.
-				if trade.wallets.fromAsset.ID == waID {
-					for _, match := range trade.matches {
-						script := match.MetaData.Proof.Script
-						if len(script) == 0 {
-							continue
-						}
-						addr, err := wallet.RefundAddress(script)
-						if err != nil {
-							return err
-						}
-						if err := ownsAddr(addr); err != nil {
-							return err
-						}
-						// Assume all refund addresses
-						// are owned by the new
-						// wallet if one is.
-						return nil
-					}
-					// If we did not find a refund address,
-					// we cannot be sure that this is the
-					// same wallet.
-					//
-					// TODO: Implement a way to check that
-					// these accounts are the same or not.
-					maybeDifferentWallet = true
+				if trade.wallets.toAsset.ID == waID || trade.wallets.fromAsset.ID == waID {
+					hasActiveTrade = true
+					break
```

Comment:
Without the `RefundAddress` work going on, we can reduce this with `hasActiveAssetOrders`:

```go
			if dc.hasActiveAssetOrders(wallet.AssetID) {
				hasActiveTrade = true
				break
			}
```

Reply:
Using.